README.md

```diff
@@ -1,5 +1,52 @@
 TaskCluster AWS Provisioner
-============================
-Bug John to write this!
+===========================
 
+The AWS Provisioner is responsible for starting Amazon EC2 instances to perform
+tasks in the TaskCluster queue.  It monitors queue lengths and uses spot
+bidding to maximize the cost-effectiveness of its resources.
 
+Hacking AWS Provisioner
+-----------------------
+
+You will need Node 4 to run the AWS provisioner.  With this in place, a simple
+`npm install` should fetch the prerequisites.
+
+### Testing
+
+To test, you will need a set of credentials.  The best way to find these is to
+ask another developer for a copy of theirs.  The configuration should look
+something like this:
+
+```
+{
+  "aws": {
+    "accessKeyId": "fake",
+    "secretAccessKey": "fake"
+  },
+  "pulse": {
+    "username": "...",
+    "password": "..."
+  },
+  "azure": {
+    "accountName":        "taskclusterdev",
+    "accountKey":         "..."
+  }
+}
+```
+
+The `aws` configuration must be present, but its value is ignored.  You can
+create your own `pulse` credentials at https://pulseguardian.mozilla.org.
+You'll need to get the Azure configuration from another TaskCluster developer.
+
+### Running Locally
+
+To run the provisioner locally, you will need a similar set of configuration.
+Then run
+
+```
+npm compile && DEBUG=* node lib/server.js development
+```
+
+Note that this is not usually the best way to test the provisioner.  Be careful
+that you manually destroy any resources the provisioner creates while it is
+running (EC2 instances or spot bids).
```

Comment:
Thanks for working on this @djmitche! I like what I see! Could you also add:

* a section on how to deploy the provisioner (e.g. if it happens automatically with a push to master, or if you need to push to heroku) and what environment variables need to be configured when setting this up
* information about what objects the provisioner creates in AWS, so people know what to clean up, and e.g. if someone were to set up a CI running in travis, they'd know how to do it in a way that they wouldn't pollute our aws environment

I think those are the only two pieces I miss. Thanks again for picking this up!